dev-docs/source/MVPTutorial/CalculatorExample/View.py

```diff
@@ -1,97 +1,82 @@
+        self._button = QPushButton("Calculate", self)
+        self._button.setStyleSheet("background-color:lightgrey")
+        grid.addWidget(self._button)
 
-        # Connect button click handler method to the button's 'clicked' signal
-        self.button.clicked.connect(self.btn_click)
-        # connect method to handle combo box selection changing to the corresponding signal
-        self.combo["display"].currentIndexChanged.connect(self.display_changed)
+        self._button.clicked.connect(self._button_clicked)
+        self._combo["display"].currentIndexChanged.connect(self._display_changed)
 
-        # Set the layout for the view widget
         self.setLayout(grid)
 
-    # The next two methods handle the signals connected to in the presenter
-    # They emit custom signals that can be caught by a presenter
-    def btn_click(self):
-        self.btnSignal.emit()
-
-    def display_changed(self):
-        self.displaySignal.emit()
+    def subscribe_presenter(self, presenter) -> None:
+        self._presenter = presenter
 
-    # Populate view
-    def create_combo_table(self, row, col, key):
-        self.combo[key] = QtWidgets.QComboBox()
-        options = ["test"]
-        self.combo[key].addItems(options)
-        self.table.setCellWidget(row, col, self.combo[key])
-
-    # The next 5 methods update the appearance of the view.
+    def _button_clicked(self) -> None:
+        self._presenter.handle_button_clicked()
 
-    def set_options(self, key, options):
-        self.combo[key].clear()
-        self.combo[key].addItems(options)
+    def _display_changed(self) -> None:
+        self._presenter.handle_display_changed()
 
-    def set_names(self, name, row):
-        text = QtWidgets.QTableWidgetItem(name)
-        text.setFlags(QtCore.Qt.ItemIsEnabled)
-        self.table.setItem(row, 0, text)
+    def _create_combo_table(self, row: int, col: int, key: str) -> None:
+        self._combo[key] = QComboBox()
+        options = ["test"]
+        self._combo[key].addItems(options)
+        self._table.setCellWidget(row, col, self._combo[key])
 
-    # Update the view with the result of a calculation
-    def setResult(self, value):
-        self.table.setItem(4, 1, QtWidgets.QTableWidgetItem(str(value)))
+    def _set_names(self, name: str, row: int) -> None:
+        text = QTableWidgetItem(name)
+        text.setFlags(ItemIsEnabled)
+        self._table.setItem(row, 0, text)
 
-    def hide_display(self):
-        self.table.setRowHidden(4, True)
+    def set_options(self, key: str, options: List[str]) -> None:
+        self._combo[key].clear()
+        self._combo[key].addItems(options)
 
-    def show_display(self):
-        self.table.setRowHidden(4, False)
+    def set_result(self, value: float) -> None:
+        self._table.setItem(4, 1, QTableWidgetItem(str(value)))
 
-    # Finally, we have the get methods to allow the presenter to read the user's input
+    def show_display(self, show: bool) -> None:
```

Comment:
This function behaviour is reversed on arguments pass on the presenter(or the name should be hide_display). It does the opposite of its intended function.